File: Cpp/Fibonacci.cpp
#include <vector>

class Solution{
public:
    /**
     * @param n: an integer
     * @return an integer f(n)
     */
    int fibonacci(int n) {
        // write your code here
        vector<int> fib(n);
        fib[0] = 0, fib[1] = 1;
        for (int i = 2; i < n; i ++) {
            fib[i] = fib[i - 1] + fib[i - 2];
        }
        return fib[n-1];
    }
};

File: Python/Fibonacci.py
class Solution:
    # @param n: an integer
    # @return an integer f(n)
    def fibonacci(self, n):
        # write your code here
        fib = []
        fib.append(0)
        fib.append(1)
        for i in range(2, n):
            fib.append(fib[i - 1] + fib[i - 2])
        return fib[n - 1]
